mobile-app/src/firebase/api.ts

```diff
@@ -96,73 +110,102 @@ export async function fetchHabitCompletionsForParticipant({
-    ),
-  );
+export async function fetchFriendData({
+  userId,
+}: {
+  userId: string;
+}): Promise<allUsersInfoT> {
+  const allFreindData: allUsersInfoT = {};
+  const myFriendIds = await fetchFriendIds({ userId });
+  for (const friendId of myFriendIds) {
+    const userDocRef = doc(firestore, "users", friendId);
+    const userDocSnap = await getDoc(userDocRef);
+    if (userDocSnap.exists()) {
+      const userData = userDocSnap.data() as userT;
+      allFreindData[friendId] = {
+        createdAt: userData.createdAt,
+        displayName: userData.displayName,
+        username: userData.username,
+        picture: userData.picture,
+      };
+    } else {
+      console.error(`No user found with ID: ${friendId}`);
+    }
+  }
+  return allFreindData;
 }
 
-export async function fetchCommonHabits({
+export async function fetchCommonHabitIds({
   participantId,
 }: {
   participantId: string;
 }): Promise<string[]> {
   const userId = atomStore.get(currentUserIdAtom);
-  // fetch habits WHERE userId in participants AND participantId in participants
-  return Object.keys(mockHabits).filter(
-    (habitId) =>
-      mockHabits[habitId].participants[userId] &&
-      mockHabits[habitId].participants[participantId],
+  const habitsCollection = collection(firestore, "habits");
+  const userQuery = query(
+    habitsCollection,
+    where(`participants.${userId}`, "!=", null),
   );
+  const userHabitsSnap = await getDocs(userQuery);
+  const commonHabitIds: string[] = [];
+  userHabitsSnap.forEach((doc) => {
+    const habitData = doc.data();
+    if (habitData.participants[participantId]) {
+      commonHabitIds.push(doc.id);
+    }
+  });
+  return commonHabitIds;
 }
 
 export async function fetchMutualFriends({
   friendId,
 }: {
   friendId: string;
 }): Promise<allUsersInfoT> {
-  const userId = atomStore.get(currentUserIdAtom);
-  // fetch friends WHERE userId in friendships AND friendId in friendships
-  const myFriendIds = Object.values(mockFriendships)
-    .filter(
-      (friendship) =>
-        friendship.user1Id === userId || friendship.user2Id === userId,
-    )
-    .map((friendship) =>
-      friendship.user1Id === userId ? friendship.user2Id : friendship.user1Id,
-    );
+  const allMutualFriendData: allUsersInfoT = {};
+  const myFriendIds: string[] = atomStore.get(friendIdsAtom);
+  const hisFriendIds: string[] = await fetchFriendIds({ userId: friendId });
 
-  const theirFriendIds = Object.values(mockFriendships)
-    .filter(
-      (friendship) =>
-        friendship.user1Id === friendId || friendship.user2Id === friendId,
-    )
-    .map((friendship) =>
-      friendship.user1Id === friendId ? friendship.user2Id : friendship.user1Id,
-    );
+  const mutualFriendIds = myFriendIds.filter((id) => hisFriendIds.includes(id));
+  for (const mutualFriendId of mutualFriendIds) {
+    const friendInfo = await fetchUserInfo({ userId: mutualFriendId });
+    allMutualFriendData[mutualFriendId] = friendInfo;
```

Comment:
We will definitely want to change this but I think it's fine for now, we will make a separate Linear task.

If we do implement this mutual friends feature, we really only need to actually fetch the picture for like 5 people max, since we can just show 5 pictures then the +8 or whatever. We don't need to make an unbounded number of fetches for users.

We could also just make it way less complex for v1 by only showing the number of mutuals, and not showing pfps.

Or we could completely remove this feature, since it's not necessary especially for v1.

For now you can just keep it and we'll figure it out later